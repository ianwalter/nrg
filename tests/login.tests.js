const { test } = require('@ianwalter/bff')
const app = require('../examples/accounts')
const { accounts, password } = require('../examples/accounts/seeds/01_accounts')

const generalUser = accounts.find(a => a.firstName === 'General User')

test('login email required validation', async ({ expect }) => {
  const response = await app.test('/login').post({ password })
  expect(response.status).toBe(400)
  expect(response.body).toMatchSnapshot()
})

test('login password required validation', async ({ expect }) => {
  const response = await app.test('/login').post({ email: generalUser.email })
  expect(response.status).toBe(400)
  expect(response.body).toMatchSnapshot()
})

test('login with invalid credentials', async ({ expect }) => {
  const payload = { ...generalUser, password: 'thisIsNotTheRightPw' }
  const response = await app.test('/login').post(payload)
  expect(response.status).toBe(400)
  expect(response.body).toMatchSnapshot()
})

test('login', async ({ expect }) => {
  const response = await app.test('/login').post({ ...generalUser, password })
  expect(response.status).toBe(201)
  expect(response.body).toMatchSnapshot()
})
