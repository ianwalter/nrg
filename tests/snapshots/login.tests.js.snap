// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login -> Already logged in 1`] = `
Object {
  "message": "Session is already authenticated",
}
`;

exports[`Login -> Disabled user 1`] = `
Object {
  "message": "Incorrect email or password",
}
`;

exports[`Login -> Invalid credentials 1`] = `
Object {
  "message": "Incorrect email or password",
}
`;

exports[`Login -> No email 1`] = `
Object {
  "feedback": Object {
    "email": Array [
      "A valid email is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Login -> No password 1`] = `
Object {
  "feedback": Object {
    "password": Array [
      "A valid password is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Login -> Valid credentials 1`] = `
Object {
  "email": "general_test@example.com",
  "emailVerified": true,
  "firstName": "General",
  "id": 1,
  "lastName": "Test",
}
`;
